addon/record-data.js

```diff
@@ -231,15 +231,19 @@ export default class FragmentRecordData extends RecordData {
           this.fragmentData[name] = data.attributes[name];
           let serverFragment = this.serverFragments[name];
           if (serverFragment) {
-            let fragmentKeys = [];
-            if (serverFragment instanceof StatefulArray) {
-              serverFragment.setupData(data.attributes[name]);
+            if (data.attributes[name] === null) {
+              delete this.serverFragments[name];
```

Comment:
TBH I'm not familiar with the internals, hence I'm not sure if it is a proper fix 😞. At least I've run this test against v4.0 and the output is now same as before.